src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/CcdAuthServiceTest.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.JurisdictionToUserMapping;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.JURSIDICTION;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.PASSWORD;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.SERVICE_TOKEN;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.USER_CREDS;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.USER_DETAILS;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.USER_ID;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.USER_NAME;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.USER_TOKEN;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class CcdAuthServiceTest {
+    @Mock
+    private AuthTokenGenerator tokenGenerator;
+    @Mock
+    private IdamClient idamClient;
+    @Mock
+    private JurisdictionToUserMapping users;
+
+    private CcdAuthService service;
+
+    @Before
+    public void before() {
+        service = new CcdAuthService(tokenGenerator, idamClient, users);
+        when(users.getUser(eq(JURSIDICTION))).thenReturn(USER_CREDS);
+        when(tokenGenerator.generate()).thenReturn(SERVICE_TOKEN);
+        when(idamClient.authenticateUser(eq(USER_NAME), eq(PASSWORD))).thenReturn(USER_TOKEN);
+        when(idamClient.getUserDetails(USER_TOKEN)).thenReturn(USER_DETAILS);
+    }
+
+    @Test
+    public void should_sucessfully_return_authInfo() {
```

Comment:
what is this testing?

Reply:
It checks the AuthApi is getting passed the correct information and the co-ordination of the users, tokenGenerator, idamClient is correctly processed.

note the @before is very specific on the information for the mock responses so it will only pass if it has been correctly called.